src/MOI_wrapper.jl

```diff
@@ -0,0 +1,355 @@
+using LinearAlgebra
+import MathOptInterface as MOI
+using ManifoldsBase
+using ManifoldDiff
+
+include("qp_block_data.jl")
+
+const _FUNCTIONS = Union{
+    MOI.VariableIndex,MOI.ScalarAffineFunction{Float64},MOI.ScalarQuadraticFunction{Float64}
+}
+
+"""
+    struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+        manifold::M
+    end
+
+Representation of points of `manifold` as a vector of `R^n` where `n` is
+`MOI.dimension(VectorizedManifold(manifold))`.
+"""
+struct VectorizedManifold{M} <: MOI.AbstractVectorSet
+    manifold::M
+end
+
+function MOI.dimension(set::VectorizedManifold)
+    return prod(ManifoldsBase.representation_size(set.manifold))
+end
+
+struct _EmptyNLPEvaluator <: MOI.AbstractNLPEvaluator end
```

Comment:
Though this is internal – maybe still add a doctoring what it is introduced for.

Reply:
Removed